### PR TITLE
Add Integration System domain and GQL types

### PR DESCRIPTION
**Pull request by tgorgol**
**Description**

Changes proposed in this pull request:

- Add Integration System domain to Director
- Add Integration System types to gql schema
- Add `integration_system:read` and `integration_system:write` scopes

**Related issue(s)**

#311

### CODE REVIEW

File: components/director/pkg/graphql/schema.graphql [hunk 1]
```diff
@@ -399,6 +403,17 @@ type HealthCheckPage implements Pageable {
 	totalCount: Int!
 }
 
+type IntegrationSystem {
```
Comment (aszecowka) [inline components/director/pkg/graphql/schema.graphql]: IntegrationSystem should also have 
`auths: [SystemAuth!]!`

File: components/director/internal/domain/integrationsystem/service.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		if repo.IsNotUnique(err) {
+			return "", errors.New("Integration System name is not unique")
+		}
+		return "", err
+	}
+
+	return id, nil
+}
+
+func (s *service) Get(ctx context.Context, id string) (*model.IntegrationSystem, error) {
```
Comment (pkosiec) [inline components/director/internal/domain/integrationsystem/service.go]: Please also add the following method: `Exist(ctx context.Context, id string) (bool, error)`

File: components/director/internal/domain/integrationsystem/resolver.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+	if err != nil {
+		return nil, err
+	}
+	defer r.transact.RollbackUnlessCommited(tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	is, err := r.intSysSvc.Get(ctx, id)
+	if err != nil {
+		return nil, err
```
Comment (pkosiec) [inline components/director/internal/domain/integrationsystem/resolver.go]: Check if it is not found - if so, return `nil, nil`

**Pull request merged**